llvm/include/llvm/SYCLLowerIR/ModuleSplitter.h

```diff
@@ -140,7 +144,14 @@ class ModuleDesc {
   Properties Props;
 
   ModuleDesc(std::unique_ptr<Module> &&M, StringRef Name = "TOP-LEVEL")
-      : M(std::move(M)), IsTopLevel(true), Name(Name) {}
+      : M(std::move(M)), IsTopLevel(true), Name(Name) {
+    // DeviceLib module doesn't include any entry point,it can be constructed
+    // using ctor without any entry point related parameter.
+    for (auto Fn : SYCLDeviceLibs) {
+      if (Name == std::string(Fn))
```

Comment:
```suggestion
      if (Name == StringRef(Fn))
```

There is no need to allocate heap memory and make a string copy for a sake of a simple comparison

Reply:
Done.
Thanks very much.